src/Plugin/RulesAction/RulesComponentAction.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesAction\RulesComponentAction.
+ */
+
+namespace Drupal\rules\Plugin\RulesAction;
+
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a generic 'Execute Rules component' action.
+ *
+ * @RulesAction(
+ *   id = "rules_component",
+ *   deriver = "Drupal\rules\Plugin\RulesAction\RulesComponentActionDeriver"
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class RulesComponentAction extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The storage of rules components.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * The ID of the rules component config entity.
+   *
+   * @var string
+   */
+  protected $componentId;
```

Comment:
I don't think so because we expose the component ID already in the action plugin defintion. The derivative should not expose the component ID as context definition.

Or do you think we should not implement this with derivatives? We could also have only once action plugin that takes the component ID as context.